### ISSUE TITLE
Enum: Value 'self.value' is unsubscriptable (unsubscriptable-object) where value is a tuple

### ISSUE THREAD

**Issue by geckon** [PyCQA/pylint#2063]
### Steps to reproduce
```
#! python3

from enum import Enum

class MyEnum(Enum):
    FOO = (1, "f")
    BAR = (2, "b")

    def __str__(self):
        return self.value[1]
```

Run `pylint` on the file above.

### Current behavior
`E: 10,15: Value 'self.value' is unsubscriptable (unsubscriptable-object)`

### Expected behavior
No error - `value` here is a tuple (and can't be reassigned).

### pylint --version output
```
$ pylint --version
No config file found, using default configuration
pylint 1.8.4, 
astroid 1.6.3
Python 3.4.6 (default, Mar 22 2017, 12:26:13) [GCC]
```

**Comment by PCManticore** [comment]
Thank you, I can reproduce the issue!
The source of problems for `enum` comes from our astroid brain tip for the enum library (https://github.com/PyCQA/astroid/blob/master/astroid/brain/brain_namedtuple_enum.py#L199), which is somewhat lossy in some areas as this one (if I recall correctly we don't fully reuse the values provided in the enum).

**Comment by sadeghipakijoo** [comment]
just make this change:
```
    def __str__(self):
        return str(self.value)[1]
```

**Comment by bje-** [comment]
I am also seeing this now after an upgrade. The message is:

```
test.py:5: [E1136(unsubscriptable-object), ] Value 'x.shape' is unsubscriptable
```

Test case:
```
"""Test case."""
import numpy as np

x = np.zeros(10, 10)
y = x.shape[0]
```

**Comment by hippo91** [comment]
@bje- in fact it is a totally different problem than the one reported by @geckon .
No need to open a new issue anyway, because it is the same as #3139.

**Comment by aakash77** [comment]
I was facing the same issue, after providing the type of the value it fixed my issue. So something like:

```python
class MyEnum(Enum):
    value: Tuple[int, str]
    FOO = (1, "f")
    BAR = (2, "b")

    def __str__(self):
        return self.value[1]
```

**Comment by hippo91** [comment]
@aakash77 thanks for the hint!

**Comment by mbyrnepr2** [comment]
This no longer appears to be an issue on latest main:
```python
pylint 2.14.0-dev0
astroid 2.11.4
Python 3.9.10 (v3.9.10:f2f3f53782, Jan 13 2022, 17:02:14) 
[Clang 6.0 (clang-600.0.57)]
```
